src/traces/waterfall/attributes.js

```diff
@@ -0,0 +1,113 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var barAttrs = require('../bar/attributes');
+var lineAttrs = require('../scatter/attributes').line;
+var extendFlat = require('../../lib/extend').extendFlat;
+
+function directionAttrs() {
+    return {
+        editType: 'style',
+        color: barAttrs.marker.color,
+        opacity: barAttrs.marker.opacity,
+        line: {
+            editType: 'style',
+            color: barAttrs.marker.line.color,
+            width: barAttrs.marker.line.width
+        }
+    };
+}
+
+module.exports = {
+
+    measure: {
+        valType: 'data_array',
+        dflt: [],
+        role: 'info',
+        editType: 'calc',
+        description: [
+            'An array containing types of values.',
+            'By default the values are considered as \'relative\'.',
+            'However; it is possible to use \'total\' to compute the sums.',
+            'Also \'absolute\' could be applied to reset the computed total',
+            'or to declare an initial value where needed.'
+        ].join(' ')
+    },
+
+    base: {
+        valType: 'number',
+        dflt: null,
+        arrayOk: false,
+        role: 'info',
+        editType: 'calc',
+        description: [
+            'Sets where the bar base is drawn (in position axis units).'
```

Comment:
What does `base` do? I suspect it simply offsets the whole trace along the value axis?

Reply:
Yes exactly. Here is a [codepen(https://codepen.io/MojtabaSamimi/pen/oVaYbX?editors=0010) with `base`.

Reply:
Great. It makes sense to have `base` as `arrayOk:false` for waterfalls :ok_hand: